heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraHeronShellController.java

```diff
@@ -74,26 +74,38 @@ public boolean killJob() {
   public boolean restart(Integer containerId) {
     // there is no backpressure for container 0, delegate to aurora client
     if (containerId == null || containerId == 0) {
-      cliController.restart(containerId);
+      return cliController.restart(containerId);
     }
 
     if (stateMgrAdaptor == null) {
       LOG.warning("SchedulerStateManagerAdaptor not initialized");
       return false;
     }
 
-    int index = containerId - 1; // stmgr container starts from 1
-    StMgr contaienrInfo = stateMgrAdaptor.getPhysicalPlan(topologyName).getStmgrs(index);
-    String host = contaienrInfo.getHostName();
-    int port = contaienrInfo.getShellPort();
+    String host = "x";
+    int port = -1;
+    int offset = "stmgr-".length();
```

Comment:
Suggest to move this logic into a function and compare the names in the reversed way:

StMgr searchContainer(Integer id) {
  prefix = "stmgr-" + id;
  for (StMgr sm : stateMgrAdaptor.getPhysicalPlan(topologyName).getStmgrsList()) {
    if (sm.getId().startsWith(prefix)) return sm;
  }
  return null;
}

It is more readable and easier to test the function.

Reply:
done